spec/unit/shoulda/matchers/active_record/association_matcher_spec.rb

```diff
@@ -646,8 +646,28 @@ def having_many_non_existent_class(model_name, assoc_name, options = {})
     end
 
     it 'accepts an association with a valid :dependent option' do
-      expect(having_one_detail(dependent: :destroy)).
-        to have_one(:detail).dependent(:destroy)
+      dependent_options(Rails.version).each do |option|
+        expect(having_one_detail(dependent: option)).
+          to have_one(:detail).dependent(option)
+      end
+    end
+
+    it 'accepts any dependent option if true' do
+      dependent_options(Rails.version).each do |option|
+        expect(having_one_detail(dependent: option)).
+          to have_one(:detail).dependent(true)
+      end
+    end
+
+    it 'rejects any dependent options if false' do
+      dependent_options(Rails.version).each do |option|
+        expect(having_one_detail(dependent: option)).
+          to_not have_one(:detail).dependent(false)
+      end
+    end
+
+    it 'accepts a nil dependent option if false' do
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.